File: src/main/java/io/quarkiverse/test/junit/mockk/InjectSpy.java
package io.quarkiverse.test.junit.mockk;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * When used on a field of a test class, the field becomes a Mockk spy,
 * that is then used to spy on the normal scoped bean which the field represents
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface InjectSpy {
}

File: src/main/kotlin/io/quarkiverse/test/junit/mockk/InjectSpy.kt
package io.quarkiverse.test.junit.mockk

/**
 * When used on a field of a test class, the field becomes a Mockk spy,
 * that is then used to spy on the normal scoped bean which the field represents
 */
@kotlin.annotation.Target(AnnotationTarget.FIELD)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class InjectSpy()
